bundles/org.openhab.binding.sensorpush/src/main/java/org/openhab/binding/sensorpush/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,438 @@
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@link CloudBridgeHandler} is responsible for communicating with the SensorPush cloud.
+ *
+ * @author Bob Adair - Initial contribution
+ */
+@NonNullByDefault
+public class CloudBridgeHandler extends BaseBridgeHandler {
+
+    private static final String AGENT = "openHAB SensorPush Binding";
+    private static final String API_URL = "https://api.sensorpush.com";
+    private static final int POLL_MIN = 2;
+    private static final int POLL_MAX = 60;
+    private static final int TIMEOUT_MIN = 1;
+    private static final int TIMEOUT_MAX = 120;
+
+    private final Logger logger = LoggerFactory.getLogger(CloudBridgeHandler.class);
+
+    private final HttpClient httpClient;
+    private final Gson gson;
+
+    private @Nullable SensorPushDiscoveryService discoveryService;
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private CloudBridgeConfiguration config = new CloudBridgeConfiguration();
+
+    private @Nullable String accessToken;
+    private @Nullable JwtInfo accessTokenInfo;
+
+    private boolean pollSensorsRun = true;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+        gson = new GsonBuilder().create();
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(SensorPushDiscoveryService.class);
+    }
+
+    public void setDiscoveryService(SensorPushDiscoveryService discoveryService) {
+        this.discoveryService = discoveryService;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(CloudBridgeConfiguration.class);
+
+        if (config.user.isBlank() || config.password.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "user/password not configured");
+            return;
+        }
+        logger.trace("CloudBridge for user {} initializing.", config.user);
+
+        if (config.poll < POLL_MIN) {
+            config.poll = POLL_MIN;
+        } else if (config.poll > POLL_MAX) {
+            config.poll = POLL_MAX;
+        }
+
+        if (config.timeout < TIMEOUT_MIN) {
+            config.timeout = TIMEOUT_MIN;
+        } else if (config.poll > TIMEOUT_MAX) {
+            config.timeout = TIMEOUT_MAX;
+        }
```

Comment:
You can specify the min/max values in the XML file. Then, the framework takes care these are in the correct range.

Reply:
I have defined min/max values in the XML ,but as I recall those constraints are only used by the UI. I thought if you defined a thing with a text file (although I don't recommend that for this binding) the framework wouldn't check the values.

Reply:
I'm fairly confident that this is also checked when using textual configuration. Would be awesome if you could try it out.

Reply:
I finally got around to testing this, and as I thought it is not checked for text configs. The min/max values in the XML are only used by the GUI. It happily let me violate the limits in a text config file by specifying poll=1 when min="2" is set for the poll parameter in the XML.

Reply:
That sounds like a bug or at least missing feature to me. Can you file an issue here? https://github.com/openhab/openhab-core/issues

Reply:
I thought I remembered from way back it being documented somewhere that those limits were only enforced in the GUI, but looking through the developer docs now I don't see it mentioned anywhere. I can create a feature request under openhab-core, but I'm wondering if a doc update would be more immediately useful. Assuming I'm not just missing something.

Reply:
I think it should be clarified first if it's a bug or a missing feature before adding the fact to the documentation. If it turns out to be a missing feature it would be good to document it.

Reply:
I've created https://github.com/openhab/openhab-core/issues/2674 for this issue.